File: problem2-cpp/challenge.cpp
/*
 * GOAL: Use the clue from Problem 1 to perform vector operations
 *
 * The program should:
 * 1. Take the positions from Problem 1's clue
 * 2. Use these as indices to access a vector
 * 3. Sum the values at these positions
 * 4. Generate a clue combining "FIBONACCI" and the sum
 *
 * Expected clue output: FIBONACCI_160
 */

#include <iostream>
#include <vector>
#include <string>

using namespace std;

class VectorProcessor
{
private:
    vector<int> data;
    vector<int> cluePositions;

public:
    VectorProcessor()
    {
        data = {10, 20, 30, 40, 50, 60, 70, 80};

        cluePositions = {0, 2, 4, 6}; //Use clue from Problem 1 here 
    }

    int processClue()
    {
        cout << "Using clue from Problem 1: [";
        for (int i = 0; i < cluePositions.size(); i++)
        {
            cout << cluePositions[i];
            if (i < cluePositions.size() - 1)
                cout << ", ";
        }
        cout << "]" << endl;

        cout << "Vector: [";
        for (int i = 0; i < data.size(); i++)
        {
            cout << data[i];
            if (i < data.size() - 1)
                cout << ", ";
        }
        cout << "]" << endl;

        cout << "Values at clue positions: ";
        int sum = 0;
        for (int i = 0; i < cluePositions.size(); i++)
        {
            int pos = cluePositions[i];
            if (pos >= 0 || pos < data.size())
            { 
                cout << data[pos];
                if (i < cluePositions.size() - 1)
                    cout << ", ";
                sum += data[pos]; 
            }
        }
        cout << endl;

        return sum;
    }

    string generateClue(int sum)
    {
        return "FIBONACCI_" + to_string(sum);
    }
};

int main()
{
    VectorProcessor processor; 

    int result = processor.processClue();
    cout << "Sum of values: " << result << endl;

    string nextClue = processor.generateClue(result);
    cout << "Clue for Python problem: " << nextClue << endl;

    return 0;
}

/*
SOLUTION - PASTE YOUR CLUE HERE:
Member Name: Aryan
Clue for Python problem: FIBONACCI_160
*/

File: problem4-final/Challenge.java
/*
 * GOAL: Parse the clue from Problem 3 and complete the final sorting challenge
 * 
 * The program should:
 * 1. Parse the clue from Problem 3 to extract the array
 * 2. Sort the array using a custom bubble sort algorithm
 * 3. Display the final success message for the team
 * 
 * Expected final output: Team completion message
 */

import java.util.Arrays;
import java.util.ArrayList;
import java.util.List;

public class Challenge {
    
    public static int[] parseClueArray(String clue) {
        try {
            int startIndex = clue.indexOf('[');
            int endIndex = clue.indexOf(']');
            
            if (startIndex == -1 && endIndex == -1) {  
                return new int[0];
            }
            
            String arrayStr = clue.substring(startIndex + 1, endIndex);
            String[] parts = arrayStr.split(",");
            
            int[] result = new int[parts.length];
            for (int i = 0; i < parts.length; i++) {
                result[i] = Integer.parseInt(parts[i].trim());
            }
            
            return result;
        } catch (Exception e) {
            return new int[0];
        }
    }
    
    public static void bubbleSort(int[] arr) {
        int n = arr.length;
        
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {  
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }
    
    public static void displayFinalSolution() {
        System.out.println("FINAL SOLUTION: The team successfully solved all 4 problems!");
        System.out.println("Team completed: C → C++ → Python → Java relay!");
        System.out.println("Congratulations 🎉! You navigated the repo, decoded the nonsense, and came out version-controlled. Respect.");
    }
    
    public static void main(String[] args) {
        String previousClue = "JAVA_SORT_[0,1,1,2,3,5,8,13]"; //Use clue from Problem 3
        
        System.out.println("Using clue from Problem 3: " + previousClue);
        
        int[] extractedArray = parseClueArray(previousClue);
        System.out.println("Extracted array: " + Arrays.toString(extractedArray));
        
        bubbleSort(extractedArray);
        System.out.println("Array after custom sort: " + Arrays.toString(extractedArray));
        
        displayFinalSolution();
    }
}

/*
SOLUTION - PASTE YOUR FINAL SOLUTION HERE:
Member Name: Aryan
Final Solution Message: [0, 1, 1, 2, 3, 5, 8, 13]
*/
